test/ha/activator_test.go

```diff
@@ -89,7 +93,13 @@ func TestActivatorHA(t *testing.T) {
 	}
 	activatorPod := pods.Items[0].Name
 
-	origEndpoints, err := getPublicEndpoints(t, clients, resourcesScaleToZero.Revision.Name)
+	publicEndpointsOpts := metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s,%s=%s",
+			serving.RevisionLabelKey, resourcesScaleToZero.Revision.Name,
+			networking.ServiceTypeKey, networking.ServiceTypePublic,
+		),
+	}
+	origEndpoints, err := pkgTest.GetEndpoints(clients.KubeClient, test.ServingNamespace, publicEndpointsOpts)
```

Comment:
This doesn't need a listOption either I think. You can use the `ServiceName` from the revision's status to fetch the public service without having to list stuff.

Reply:
OK. I'll look into this. Thanks

Reply:
I can see that the ServiceName status field of the revision always has the same value as the revision's name. Can this ever be different? I could use directly the revision name.